### PR TITLE
feat(frontend): array of resources

### PR DESCRIPTION
**Pull request by fatonramadani**
Screenshot 2024-07-17 at 15 25 54

Screenshot 2024-07-17 at 15 45 40

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit ae1bb4074d0f30b7bae4d1eba614fdfbf428132e  | 
|--------|--------|

### Summary:
Added support for handling 'resource' type in arrays across multiple frontend components and functions.

**Key points**:
- Updated `SchemaProperty` in `frontend/src/lib/common.ts` to include `resource` type and `resourceType` field.
- Modified `ArgInput.svelte`, `ArrayTypeNarrowing.svelte`, `LightweightArgInput.svelte`, and `PropertyEditor.svelte` to handle `resource` type in arrays.
- Added `ResourcePicker` and `LightweightResourcePicker` components for selecting resources.
- Updated `argSigToJsonSchemaType` function in `frontend/src/lib/inferArgSig.ts` to infer `resource` type in arrays.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: frontend/src/lib/inferArgSig.ts [hunk 1]
```diff
@@ -102,12 +102,24 @@ export function argSigToJsonSchemaType(
...
+
+			if (t.list && typeof t.list == 'object' && 'resource' in t.list && t.list.resource) {
+				resource = t.list.resource as string
+			}
+
+			if (resource) {
+				newS.items = { type: 'resource', resource: resource }
+			} else {
+				newS.items = { type: 'object' }
+			}
```
Comment (rubenfiszel) [inline frontend/src/lib/inferArgSig.ts]: seems convoluted, why not just add more cases to the top elseif

File: frontend/src/lib/components/schema/PropertyEditor.svelte [hunk 2]
```diff
@@ -34,9 +34,10 @@
 	export let order: string[] | undefined = undefined
 	export let itemsType:
 		| {
-				type?: 'string' | 'number' | 'bytes' | 'object'
+				type?: 'string' | 'number' | 'bytes' | 'object' | 'resource'
 				contentEncoding?: 'base64'
 				enum?: string[]
+				resource?: string
```
Comment (rubenfiszel) [inline frontend/src/lib/components/schema/PropertyEditor.svelte]: resourceType would be more consistent

Comment (rubenfiszel) [reply frontend/src/lib/components/schema/PropertyEditor.svelte]: but maybe not

Comment (fatonramadani) [reply frontend/src/lib/components/schema/PropertyEditor.svelte]: Done

File: frontend/src/lib/inferArgSig.ts (hunk 1 above)
Comment (fatonramadani) [reply frontend/src/lib/inferArgSig.ts]: Done

**Pull request merged**